### PR TITLE
Weaken constraints

### PR DESCRIPTION
**Pull request by rossabaker**

### CODE REVIEW

File: servlet/src/main/scala/org/http4s/servlet/Http4sServlet.scala [hunk 1]
```diff
@@ -42,8 +43,17 @@ abstract class Http4sServlet[F[_]](
     service: HttpApp[F],
     servletIo: ServletIo[F],
     dispatcher: Dispatcher[F],
-)(implicit F: Async[F])
+)(implicit F: Sync[F])
     extends HttpServlet {
+
+  @deprecated("Binary compatibility", "0.23.12")
+  private[servlet] def this(
```
Comment (rossabaker) [inline servlet/src/main/scala/org/http4s/servlet/Http4sServlet.scala:50]: @armanbilge, do I recall that you discovered a trick for the static forwarder on these?  This is the one that's going to fail.

Comment (armanbilge) [reply servlet/src/main/scala/org/http4s/servlet/Http4sServlet.scala:50]: This is a constructor, so it shouldn't have a static forwarder right?

Comment (rossabaker) [reply servlet/src/main/scala/org/http4s/servlet/Http4sServlet.scala:50]: Shit. I linked the wrong one.  I meant [this one](https://github.com/http4s/http4s-servlet/pull/24/files#diff-4e297c4945c6c3b66ce3d74472f332e6d0f5a79a56ee483c2a3f6dafc57f0c2aR104).

Comment (rossabaker) [reply servlet/src/main/scala/org/http4s/servlet/Http4sServlet.scala:50]: That link isn't being very useful.  Line 104 of `BlockingHttp4sServlet`.

File: servlet/src/main/scala/org/http4s/servlet/BlockingHttp4sServlet.scala [hunk 2]
```diff
@@ -88,4 +100,12 @@ object BlockingHttp4sServlet {
       DefaultServiceErrorHandler,
       dispatcher,
     )
+
+  protected[servlet] def apply[F[_]](
```
Comment (armanbilge) [inline servlet/src/main/scala/org/http4s/servlet/BlockingHttp4sServlet.scala:104]: This one? I think your only recourse to preserve static forwarders is to leave it public and rely on deprecation and source-incompatible changes to dissuade use, as you've done here. Not sure what `protected` does for `object` methods.

Comment (armanbilge) [reply servlet/src/main/scala/org/http4s/servlet/BlockingHttp4sServlet.scala]: Note that in Scala 3 it's the other way around. Even package-private object methods will generate static-forwarders.
- https://github.com/lampepfl/dotty/issues/10842

This causes problems with MiMa for internal methods, however.
- https://github.com/lightbend/mima/issues/694

Comment (rossabaker) [reply servlet/src/main/scala/org/http4s/servlet/BlockingHttp4sServlet.scala]: Fuck.  I'm just going to leave that one alone then.  They can use the constructor, and Servlet > 3 is coming.

Comment (armanbilge) [reply servlet/src/main/scala/org/http4s/servlet/BlockingHttp4sServlet.scala]: What you have here is fine for Scala 2 and 3. The Scala 3 thing works in our favor here, but bites in different situations until MiMa can fix it.

**Comment by rossabaker** [comment]
I think maybe the sbt-http4s-org upgrade on the merge commit is enforcing more things, but I don't know why it wasn't already broken?  The last three commits are odd.

**Comment by armanbilge** [comment]
Ah, it's because you were seeing build failures on the "push" jobs, not the "merge" jobs. I think this PR was based against a bad branch.

**Comment by rossabaker** [comment]
Yeah, branches were in flux as I was doing this.

**Pull request merged**